src/facet_display/Facet.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react"
+import { contains } from "ramda"
+
+import { SearchFacetItem } from "./SearchFacetItem"
+import { BucketWithLabel } from "./FacetDisplay"
+
+const MAX_DISPLAY_COUNT = 5
+const FACET_COLLAPSE_THRESHOLD = 15
+
+interface Props {
+  name: string
+  title: string
+  results: BucketWithLabel[] | null
+  currentlySelected: string[]
+  onUpdate: React.ChangeEventHandler<HTMLInputElement>
+  expandedOnLoad: boolean
+}
+
+function SearchFacet(props: Props) {
+  const { name, title, results, currentlySelected, onUpdate, expandedOnLoad } =
+    props
+
+  const [showFacetList, setShowFacetList] = useState(expandedOnLoad)
+  const [showAllFacets, setShowAllFacets] = useState(false)
+
+  const titleLineIcon = showFacetList ? "arrow_drop_down" : "arrow_right"
+
+  return results && results.length === 0 ? null : (
+    <div className="facets mb-3">
+      <button
+        className="filter-section-button pl-3 py-2 pr-0"
+        type="button"
+        aria-expanded={showFacetList ? "true" : "false"}
+        onClick={() => setShowFacetList(!showFacetList)}
+      >
+        {title}
+        <i className={`material-icons ${titleLineIcon}`} aria-hidden="true">
+          {titleLineIcon}
+        </i>
+      </button>
+      {showFacetList ? (
+        <React.Fragment>
+          {results ?
+            results.map((facet, i) =>
+              showAllFacets ||
+                i < MAX_DISPLAY_COUNT ||
+                results.length < FACET_COLLAPSE_THRESHOLD ? (
+                  <SearchFacetItem
+                    key={i}
+                    facet={facet}
+                    isChecked={contains(facet.key, currentlySelected || [])}
+                    onUpdate={onUpdate}
+                    name={name}
+                    displayKey={facet.label ? facet.key : facet.key}
```

Comment:
```suggestion
                    displayKey={facet.label ? facet.label : facet.key}
```

Alternatively... Optional properties are sometimes annoying. Could just make label required, and have  `resultsWithLabels` set `label` to `key` if a better label isn't found.